src/pages/index.js

```diff
@@ -161,22 +229,81 @@ function fillProfileForm() {
 // Function to handle edit profile form submission
 function handleEditProfileFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = nameInput.value;
-  profileVocation.textContent = vocationInput.value;
-  operatePopup(editModal);
+
+  const submitButton = evt.submitter;
+
+  submitButton.textContent = "Saving...";
+
+  // Get user inputs
+  const updatedData ={
+    name: nameInput.value,
+    about: vocationInput.value,
+  }
+
+  // Call API to update user info
+  api.editUserInfo(updatedData)
+  .then((user) => {
+    profileName.textContent = user.name;
+    profileVocation.textContent = user.about;
+
+    operatePopup(editModal);
+  })
+  .catch((error) => {
+    console.error(error);
+  })
+  .finally(() => {
+    submitButton.textContent = "Save";
+  });  
+}
+
+function handleNewAvatarFormSubmit(evt) 
+{
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  submitButton.textContent = "Saving...";
+
+  // Call the API to update user Avatar
+  api.editUserAvatar(avatarInput.value)
+    .then((user) => {
+      avatarElement.src = user.avatar;
+      evt.target.reset();
```

Comment:
* Please, disable the `avatar` save button as well after submission


`disableButton` should be used here as well

Reply:
got the button to disable after submission